### ISSUE TITLE
Waiting for reboot in Ansible STILL does not work

### ISSUE THREAD

**Issue by davidfokkema** [HiSPARC/pysparc#25]
I installed a new Raspberry Pi. I got the following error:
```
NOTIFIED: [wait for server to come back] **************************************
fatal: [pysparc1.his -> 127.0.0.1] => One or more undefined variables: 'ansible_ssh_host' is undefined
```
I'm getting sick of this. Fix this once and for all.

**Comment by 153957** [comment]
Is there any help in this discussion: ansible/ansible#10616 ?

**Comment by davidfokkema** [comment]
Maybe, I'll look into it when I have more time.

**Comment by gtaylor** [comment]
Seeing something slightly different:

Workaround that used to work on 1.9
```
- name: reboot server to get the new extras kernel modules
  shell: sleep 2 && shutdown -r now "Ansible updates triggered"
  async: 1
  poll: 0
  ignore_errors: true
```
The output
```
fatal: [xxx.xx.xx.xxx]: FAILED! => {"changed": false, "failed": true, "msg": "The async task did not return valid JSON: No JSON object could be decoded"}
```

**Comment by davidfokkema** [comment]
I think the problem we are seeing is that the variables ``ansible_ssh_host`` and ``inventory_hostname`` are either incosistent, or not very clearly documented. In my opinion, anyway. I tried various incantations, which always failed for *some* hosts. I have `real' hosts, and a vagrant box. It never works for all of them.

**Comment by davidfokkema** [comment]
@gtaylor the problem *you* are facing, I guess, is the value for ``async``. That is the amount of time (in seconds) that ansible will wait for your asynchronous task. In a sense, it is a timeout value. You've set it to *one* second, while your task will take at least *two* seconds (the sleep, plus the shutdown command). I guess that if you increase ``async`` to five or ten seconds, your problem goes away.

**Comment by davidfokkema** [comment]
Fixed in 56c086c1495d3c055c06643c405129dab51d6a39 and d23cc524cebab9518b4bb66c6790aac941196d25, already some time ago.
